### DISCUSSION TITLE
[15일차 - 지원] require와 import의 차이점

### DISCUSSION THREAD

**Discussion by smarfy99** [zzangCS/zzangInterview#49] [프론트 면접 질문]
require와 import의 차이점에 대해 말해보세요.

**Comment by ez1n** [comment]
- require과 import 모두 자바스크립트에서 다른 모듈을 가져오는 키워드

### 작동 방식 차이
1. **require**
- CommonJS 모듈 시스템을 따름
- 동기적으로 모듈을 가져오며 가져온 모듈을 객체로 반환
- Node.js에서 사용됨
- 변수를 다시 할당할 수 있음 → 런타임에 모듈을 가져오기 때문에 동적으로 모듈 경로 지정 가능

2. **import**
- ES6(ECAMScript 2015)에서 사용
- 비동기적으로 모듈을 가져오며 가져온 모듈을 Promise 객체로 반환
- 브라우저에서 사용됨
- export 키워드를 사용하여 객체를 내보낸다.
- Babel과 같은 ES6 코드를 변환해주는 도구가 필요함
- 변수를 다시 할당할 수 없음 → 정적으로 모듈을 가져오기 때문에 동적으로 모듈 경로 지정 불가능
